### PR TITLE
Themes: Validate 'section' prop in theme sheets

### PR DESCRIPTION
**Pull request by seear**
The route for a theme details page is:
`/theme/:slug/:section?/:site?`
where `section = ( details | support | documentation )`

* Validate the section prop at the sheet component level, assuming `details` if invalid
* Add a 'details' section when constructing internal links

**To Test**
1) Go to http://calypso.localhost:3000/design/ and check that the _Details_ option from the ... button for a theme brings up the new Calypso theme sheet
* For _all sites_ (or logged-out) the url should be `http://calypso.localhost:3000/theme/{theme}/details`
* For single site, the url should be `http://calypso.localhost:3000/theme/{theme}/details/{site}`

2) Manually enter a url such as `http://calypso.localhost:3000/theme/mood/blah` and check that the sheet displays with the *details* section highlighted

3) With `manage/themes/details` disabled in `config/development.json` check that the details button links to the old themes showcase

### CODE REVIEW

**Comment by ockham** [comment]
Aside, doesn't the data component's `section` prop [here](https://github.com/Automattic/wp-calypso/blob/b2cc08d261f47cd79adaeafe9a47be50d28fccf9/client/my-sites/themes/controller.js#L154) really belong to the `Sheet` component on the next line?

Also, what about using a default prop for `section`, and a `oneOf` `propType`? (i.e. `section: React.PropTypes.oneOf([ 'details', 'documentation', 'support' ] )`)? (Though the latter doesn't hard-validate `section`, but might still be nice to have.)

**Comment by seear** [comment]
> Aside, doesn't the data component's section prop here really belong to the Sheet component on the next line?

3275124

> Also, what about using a default prop for section

71cfc32

**Comment by ockham** [comment]
LGTM :+1:

File: client/my-sites/themes/helpers.js [hunk 1]
```diff
@@ -50,17 +50,16 @@ var ThemesHelpers = {
...
 
-		if ( site.jetpack ) {
-			return site.options.admin_url + 'themes.php?theme=' + theme.id;
+		let baseUrl = ThemesHelpers.oldShowcaseUrl + theme.id;
+		if ( config.isEnabled( 'manage/themes/details' ) ) {
+			baseUrl = `/theme/${ theme.id }/details`;
 		}
 
-		return baseUrl + `${ theme.id }/${ site.slug }`;
+		return baseUrl + ( site ? `/${ site.slug }` : '' );
```
Comment (ockham) [inline client/my-sites/themes/helpers.js]: See https://github.com/Automattic/wp-calypso/pull/390/files/2bce8561b38f6242360b1235e80d9ea98026f439#r56220368

**Comment by ockham** [comment]
Added one more note

File: client/my-sites/themes/helpers.js (hunk 1 above)
Comment (ockham) [inline client/my-sites/themes/helpers.js]: (There are a couple more occurrences of `site.slug` in this file -- might make sense to replace them, too, while we're here.)

Comment (seear) [inline client/my-sites/themes/helpers.js]: We have other occurrences dotted around, so let's do them all in one go in a separate PR and track in #4057.
